test/sdks-e2e/sdks_test.go

```diff
@@ -51,6 +54,43 @@ func (s *SDKsSuite) TestUDFunctionAndSink() {
 		VertexPodLogContains("go-udsink", "hello", PodLogCheckOptionWithContainer("udsink"), PodLogCheckOptionWithCount(3))
 }
 
+func (s *SDKsSuite) TestJavaSdk() {
```

Comment:
Do we need this test case if we add to the existing test a flatmap vertex in Java?

Reply:
I have asked @yhl25 to also add Java-SDK based flatmap to the e2e test. We also need to have the reduce tests too, as of today we do not have E2E reduce tests for any SDKs, Java will be the first one to have it.

Reply:
I think we have to rename `TestJavaSdk` to `TestReduceSDK`. WDYT?

Reply:
Oh I see. Reduce tests are needed because supporting reduce is a new feature in numaflow-java. +1

Reply:
yeah it makes sense to rename it